### ISSUE TITLE
[Messenger] auto-setup option does not support setting a boolean value.

### ISSUE THREAD

**Issue by davidbarratt** [symfony/symfony#26996]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 4.1

A config like this:
```
framework:
    messenger:
        adapters:
          rsvp:
            dsn: "%env(MESSENGER_ADAPTER_DSN)%"
            options:
              auto-setup: false
```
will not work, because Messeanger only reads string values, so the config has to be like this:
```
framework:
    messenger:
        adapters:
          rsvp:
            dsn: "%env(MESSENGER_ADAPTER_DSN)%"
            options:
              auto-setup: 'false'
```
This should be fixed so developers don't get an unexpected result.

**Comment by sroze** [comment]
Fancy sending a PR for this? Note that we need to keep the `false` string option was it's the only way we can have the XML working I reckon.

**Comment by sroze** [comment]
To be fixed in #27001
